test/Driver/checkedc-notsupported.cpp

```diff
@@ -0,0 +1,9 @@
+// Checked C extension is not supported for C++.   Make sure driver
```

Comment:
It's probably low priority, but we could also add a test case for a .c file that is forced to be compiled as C++ using command-line options.  That's a pattern I've certainly seen before.